### PR TITLE
[RISCV] Split convert to FP pseudos by SEW

### PR DESCRIPTION
**Pull request by michaelmaitland**
* [RISCV] Split narrowing convert to FP pseudos by SEW
* [RISCV] Split Widening convert to FP pseudos by SEW
* [RISCV] Split single width convert to FP pseudos by SEW

I plan to commit each commit separate once approved.

### CODE REVIEW

**Review by topperc — APPROVED** [review]
LGTM

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp [hunk 1]
```diff
@@ -17674,6 +17673,85 @@ static MachineBasicBlock *emitVFROUND_NOEXCEPT_MASK(MachineInstr &MI,
...
+  unsigned Log2SEW = MI.getOperand(RISCVII::getSEWOpNum(MI.getDesc())).getImm();
+  // There is no E8 variant for VFCVT_F_X.
+  assert(Log2SEW >= 4);
+  // Since MI (VFROUND) isn't SEW specific, we cannot use a macro to make
+  // handling of different (LMUL, SEW) pairs easier because we need to pull the
+  // SEW immediate from MI, and that information is not avaliable during macro
+  // expansion.
+  unsigned CVTFOpc;
+  if (Log2SEW == 4) {
+    switch (LMul) {
```
Comment (wangpc-pp) [inline llvm/lib/Target/RISCV/RISCVISelLowering.cpp:17686]: We may need to add a SearchTable/SearchIndex for this.

**Review by wangpc-pp — APPROVED** [review]
LGTM.

**Comment by michaelmaitland** [comment]
Closed by 2e0e3b0f10b5c433504e31ca73e55f86e0b27660, 60a1158f31bb99a5d049d5cac5a704d906a1277e, and 469493f55639a5cce2fc69300222be609fcc5238.

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp (hunk 1 above)
Comment (michaelmaitland) [reply llvm/lib/Target/RISCV/RISCVISelLowering.cpp:17686]: https://github.com/llvm/llvm-project/pull/88742

**Pull request closed without merging**